config/config-api/src/main/java/com/thoughtworks/go/config/materials/ScmMaterialConfig.java

```diff
@@ -55,6 +60,16 @@
     @ConfigAttribute(value = "autoUpdate", optional = true)
     private boolean autoUpdate = true;
 
+    @ConfigAttribute(value = "username", allowNull = true)
+    protected String userName;
```

Comment:
`allowNull` on username attribute for `TfsMaterialconfig` was set to `false` not it is `true`. This must be handled in the entity validation